nexus/reconfigurator/execution/src/clickhouse.rs

```diff
@@ -152,6 +153,36 @@ pub(crate) async fn deploy_nodes(
     Ok(())
 }
 
+pub(crate) async fn deploy_single_node(
+    opctx: &OpContext,
+    zones: &BTreeMap<SledUuid, BlueprintZonesConfig>,
+) -> Result<(), anyhow::Error> {
+    if let Some(zone) = zones
+        .values()
+        .flat_map(|zones| {
+            zones.zones.iter().find(|zone| zone.zone_type.is_clickhouse())
+        })
+        .next()
+    {
+        let admin_addr = SocketAddr::V6(SocketAddrV6::new(
+            zone.underlay_address,
+            CLICKHOUSE_ADMIN_PORT,
+            0,
+            0,
+        ));
+        let admin_url = format!("http://{admin_addr}");
+        let log = opctx.log.new(slog::o!("admin_url" => admin_url.clone()));
+        let client = ClickhouseSingleClient::new(&admin_url, log.clone());
+        client.init_db().await.map(|_| ()).map_err(|e| {
```

Comment:
It's unclear to me if this is the right place to try to do this initialization, since we'll presumably have just spun up the server. It might be better to wait until the single-node server appears in an inventory collection, like the multi-node server does. OTOH, it shouldn't take long to start an uninitialized database, so we might be able to get away with a small retry loop and some longish timeouts, which seems significantly simpler.

Reply:
You could `ping` the server and wait for that to succeed before trying this. The `init_db()` call also returns an error indicating the server isn't available, so you might be able to just retry this call itself. The SQL it applies is idempotent.